astro/src/content/docs/lifecycle/authenticate-users/contextual-multi-factor.mdx

```diff
@@ -0,0 +1,107 @@
+---
+title: Contextual Multi-Factor Authentication (MFA)
+description: Learn about how FusionAuth decides to trigger multi-factor authentication (MFA) in the login flow.
+navcategory: developer
+section: lifecycle
+subcategory: authenticate users
+tags: auth as a service
+---
+import Breadcrumb from 'src/components/Breadcrumb.astro';
+import PremiumPlanBlurb from 'src/content/docs/_shared/_premium-plan-blurb.astro';
+import EnterprisePlanBlurb from 'src/content/docs/_shared/_enterprise-plan-blurb.astro';
+import MfaDecisionTree from 'src/diagrams/docs/lifecycle/authenticate-users/mfa-decision-tree.astro';
+import TenantMfaDecisionTree from 'src/diagrams/docs/lifecycle/authenticate-users/tenant-mfa-decision-tree.astro';
+import ApplicationMfaDecisionTree from 'src/diagrams/docs/lifecycle/authenticate-users/app-mfa-decision-tree.astro';
+
+## Overview
+
+FusionAuth applies multi-factor authentication based on a number of configuration and situational behaviors. In this document you'll learn about the way FusionAuth automatically applies MFA during the login process.
+
+If you are looking for guidance on implementing MFA, please [review the MFA guide](/docs/lifecycle/authenticate-users/multi-factor-authentication).
+
+## MFA Inside The Authentication Process
+
+There are four major considerations for whether MFA is applied during the login process.
+
+* What is the MFA policy for the tenant or application? It can be either `Disabled`, `Enabled`, or `Required`.
+* Does the user have MFA enabled? If this is not the case, then an MFA challenge is never triggered, but the user may be prompted to set up MFA, based on the tenant or application MFA policy.
+* Did the user log in using an Identity Provider such as Google or OIDC? In this case, MFA is never required for the user.
+* Has the device passed appropriate contextual checks such as being on a known device?
+
+Here's a high level diagram of the logic:
+
+<MfaDecisionTree alt="High level diagram of MFA logic."/>
+
+### Contextual Checks
+
+The contextual checks are based on attributes of the request evaluated every time an authentication happens and and include:
+
+* Has this device been seen before?
+* Has this user been seen before on this device?
+* Is there a suspicious login detected for this authentication attempt?
+
+The duration of trust of the device can be configured with the `twoFactorTrustIdTimeToLiveInSeconds` tenant configuration value.
```

Comment:
Should this be an `InlineField` element? Should we give the full path `tenant.externalIdentifierConfiguration.twoFactorTrustIdTimeToLiveInSeconds`?